File: cp_rust/src/bin/abc394-b-cat.rs
// Created by Ayush Biswas at 2026/07/08 20:25
// https://atcoder.jp/contests/adt_medium_20250415_3/tasks/abc394_b
use cp_lib::*;

// @code begin
use cpio::*;

sol! {
    fn solution(
        n: usize,
        (mut s): [String]; n
    ) -> String {
        s.sort_by_key(|x| x.len());
        s.join("")
    }
}
// @code end

File: cp_nim/src/bin/abc394-b-cat.nim
# Created by Ayush Biswas at 2026/07/19 09:48
# https://atcoder.jp/contests/adt_medium_20250415_3/tasks/abc394_b
import ../lib/cpio

# @code begin

import algorithm

var r = newReader()

let n = r[int]
let res = r[seq[string], n].sortedByIt(it.len).join("")

echo res

# @code end
